### DISCUSSION TITLE
IstioOperator is discouraged, so, is a lot of documentation outdated?

### DISCUSSION THREAD

**Discussion by josemrs** [istio/istio#53369] [Q&A] [answered]
Hi,

I'm quite confused here. According to documentation the use of IstioOperator is discouraged, and [this](https://github.com/istio/istio/discussions/50352#discussioncomment-9073294) answer says "Very discouraged", fine I'm quite happy to use Helm, actually I rather use Helm.

Then, I find a lot official documentation points to IstioOperator resource to deploy things like [multi-cluster deployments](https://istio.io/latest/docs/setup/install/multicluster/primary-remote_multi-network/)

Looking outside the official documentation doesn't help either as every random tutorial seems to use IstioOperator too.

What am I missing here? is the doc "outdated"? is there a way to translate IstioOperator into a deployment using Helm?

Thanks,
Jose

**Comment by zirain** [comment, marked as answer]
the controller is deprecated, not the CRD.

**Reply by josemrs** [reply]
I understand then that IstioOperator resources are meant to be used with istioctl tool, like 

`istioctl install --set values.pilot.env.EXTERNAL_ISTIOD=true --context="${CTX_CLUSTER1}" -f cluster1.yaml`

where cluster1.yaml is the IstioOperator resource.

Is there a way to translate between IstioOperator CRD and values.yaml for pure Helm deployment?

In out deployment we cannot deploy using istioctl, we have to deploy at scale using Helm, the use of per-service tools, like istioctl, is not an option for us.

For instance, I've seen in another response that IstioOperator resource uses istio/gateway chart, this makes me wonder if I need to use istio/gateway on top of istio/istiod or just istio/gateway will do both for a multi-cluster primary-remote deployment.

**Reply by howardjohn** [reply]
For the majority of IstioOperator configurations, there is a 1:1 mapping to helm. Anything under `values:`.

Some of the stuff under `component...k8s` is not 1:1, but usually has equivilent.

We are also building out a command to translate this automatically. Feedback welcome! https://github.com/istio/istio/pull/52281

**Comment by josemrs** [comment]
To add more context we use ArgoCD to deploy all services in our clusters. ArgoCD does not even use helm install, it does helm template and then apply the rendered templates. We keep our per-cluster value files in Git so ArgoCD automatically picks the right one for each cluster.

Not sure how to do this when running istioctl is the only real option for deployment

**Comment by josemrs** [comment]
That command looks quite promising. Thanks a lot.

For things like generating cluster config for primariy-remote access, is it
a istioctl specific thing?

**Reply by howardjohn** [reply]
That is and cannot be purely expressed in helm since its basically querying/modifying resources in one cluster, and emitting config to apply to a different cluster

**Comment by josemrs** [comment]
May it be replaced by a kubeconfig (we have means for generating or
retrieve those) or using a Service Account?

**Reply by howardjohn** [reply]
Yeah you can manually generate it. Youll need to make sure you use a similar format (in a SEcret, with `istio/multiCluster` label, etc) though

**Reply by keithmattix** [reply]
Yeah I did this before in Terraform. Maybe we should add docs on istio.io to describe how to do it and promote the naming format to a stable API

**Comment by josemrs** [comment]
Great, I'll try that.

Thanks

**Comment by josemrs** [comment]
Interesting, do you terraformed the secret for the primary cluster access
to the remote?

I'm AFK but tomorrow I'll check if I can figure out the format of the
secret and how we can generate it.

Thanks, your are being really supporting.
